runtime/builtin_types.go

```diff
@@ -559,6 +560,42 @@ func builtinRange(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException)
 	return ListType.Call(f, []*Object{r}, nil)
 }
 
+func builtinRawInput(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
+	if len(args) > 1 {
+		msg := fmt.Sprintf("[raw_]input expcted at most 1 arguments, got %d", len(args))
+		return nil, f.RaiseType(TypeErrorType, msg)
+	}
+
+	fin := os.Stdin
+	fout := os.Stdout
+	if fin == nil {
+		msg := fmt.Sprintf("[raw_]input: lost sys.stdin")
+		return nil, f.RaiseType(RuntimeErrorType, msg)
+	}
+
+	if fout == nil {
+		msg := fmt.Sprintf("[raw_]input: lost sys.stdout")
+		return nil, f.RaiseType(RuntimeErrorType, msg)
+	}
+
+	in := bufio.NewReader(fin)
+	if len(args) == 1 {
+		prompt, _ := ToStr(f, args[0])
+		fmt.Fprint(fout, prompt.Value())
+	}
+	s, _, _ := in.ReadLine()
+
+	// Todo: Should implement more cases of error.
+	pySsizeTmax := ((^uint(0)) - 1) >> 1
```

Comment:
I think that if len(s) were ever > than this value that Go would panic so don't bother checking this.